From d55b1c25c9a1bc38d8f795086a6388ee385c72fc Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 26 Jan 2020 00:19:40 +0100
Subject: [PATCH 158/467] media: ov5640: Sleep after poweroff to ensure next
 poweron is not too early

It's easy to use v4l2 userspace api in such a way that user can trigger
a brownout on the sensor instead of a proper powerdown and powerup.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/media/i2c/ov5640.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/i2c/ov5640.c b/drivers/media/i2c/ov5640.c
index 2f6f97ea6..8a53c0c36 100644
--- a/drivers/media/i2c/ov5640.c
+++ b/drivers/media/i2c/ov5640.c
@@ -1971,6 +1971,7 @@ static void ov5640_set_power_off(struct ov5640_dev *sensor)
 	ov5640_power(sensor, false);
 	regulator_bulk_disable(OV5640_NUM_SUPPLIES, sensor->supplies);
 	clk_disable_unprepare(sensor->xclk);
+	msleep(100);
 }
 
 static int ov5640_set_power_mipi(struct ov5640_dev *sensor, bool on)
-- 
2.34.0

